src/python/pants/goal/run_tracker.py

```diff
@@ -105,6 +107,55 @@ def start(self, run_start_time: float, specs: List[str]) -> None:
             }
         )
 
+    def get_anonymous_telemetry_data(self, repo_id: str) -> Dict[str, str | List[str]]:
+        # TODO: Find a way to know from a goal name whether it's a standard or a custom
+        #  goal whose name could, in theory, reveal something proprietary. That's more work than
+        #  we want to do at the moment, so we maintain this manual list for now.
+        standard_goals = {
+            "count-loc",
+            "dependees",
+            "dependencies",
+            "export-codegen",
+            "filedeps",
+            "filter",
+            "fmt",
+            "lint",
+            "list",
+            "package",
+            "py-constraints",
+            "repl",
+            "roots",
+            "run",
+            "tailor",
+            "test",
+            "typecheck",
+            "validate",
+        }
+
+        def maybe_hash_with_repo_id_prefix(s: str) -> str:
+            qualified_str = f"{repo_id}.{s}" if s else repo_id
+            # If the repo_id is the empty string we return a blank string.
+            return sha256(qualified_str.encode()).hexdigest() if repo_id else ""
+
+        return {
+            "run_id": str(self._run_info.get("id", uuid.uuid4())),
+            "timestamp": str(self._run_info.get("timestamp")),
+            # Note that this method is called after the StreamingWorkunitHandler.session() ends,
+            # i.e., after end_run() has been called, so duration will be set.
+            "duration": str(self._run_total_duration),
+            "outcome": str(self._run_info.get("outcome")),
+            "platform": platform.platform(),
+            "python_implementation": platform.python_implementation(),
+            "python_version": platform.python_version(),
+            "pants_version": str(self._run_info.get("version")),
+            # Note that if repo_id is the empty string then these three fields will be empty.
+            "repo_id": maybe_hash_with_repo_id_prefix(""),
+            "machine_id": maybe_hash_with_repo_id_prefix(str(uuid.getnode())),
+            "user_id": maybe_hash_with_repo_id_prefix(getpass.getuser()),
+            # Note that we conserve the order in which the goals were specified on the cmd line.
+            "goals": [goal for goal in self.goals if goal in standard_goals],
```

Comment:
What do you think about adding a goal count that does not exclude any goals or else a "used_custom_goals" bool field?

Reply:
> used_custom_goals" bool field?

This would be excellent, gives us an anonymized approximation of how much plugins are being used.